core/src/main/java/bisq/core/trade/statistics/TradeStatisticsManager.java

```diff
@@ -150,4 +154,30 @@ private void dump() {
             jsonFileManager.writeToDisc(Utilities.objectToJson(array), "trade_statistics");
         }
     }
+
+    static class WrapperTradeStatistics2 {
+        private TradeStatistics2 tradeStatistics;
+
+        public WrapperTradeStatistics2(TradeStatistics2 tradeStatistics) {
+
+            this.tradeStatistics = tradeStatistics;
+        }
+
+        public TradeStatistics2 unwrap() {
+            return this.tradeStatistics;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj) return true;
+            if (obj == null || getClass() != obj.getClass()) return false;
+            var wrapper = (WrapperTradeStatistics2) obj;
```

Comment:
Don't you need a check that obj is actually a `WrapperTradeStatistics2` here? Could cause an exception otherwise.

Reply:
Could add it to make it safer if it is reused somewhere.
Atm using it like that should not cause this case to ever happen.
```
.map(WrapperTradeStatistics2::new)
.distinct()
```

Reply:
But will add a check to be sure

Reply:
ah wait - I've added it already 😉 
```
 if (obj == null || getClass() != obj.getClass()) return false;
```

Reply:
Oh, right, that's right, missed it.